corehq/apps/domain/models.py

```diff
@@ -921,9 +931,44 @@ def delete(self):
             include_docs=False,
         )]
         iter_bulk_delete(db, related_doc_ids, chunksize=500)
+        self._delete_web_users_from_domain()
+        self._delete_sql_objects()
         super(Domain, self).delete()
         Domain.get_by_name.clear(Domain, self.name)  # clear the domain cache
 
+    def _delete_web_users_from_domain(self):
+        from corehq.apps.users.models import WebUser
+        web_users = WebUser.by_domain(self.name)
+        for web_user in web_users:
+            web_user.delete_domain_membership(self.name)
+
+    def _delete_sql_objects(self):
+        from casexml.apps.stock.models import DocDomainMapping
+        from corehq.apps.locations.models import SQLLocation, LocationType
+        from corehq.apps.products.models import SQLProduct
+
+        cursor = connection.cursor()
+
+        # Delete stock data
+        cursor.execute(
+            "DELETE FROM stock_stocktransaction "
+            "WHERE report_id IN (SELECT id FROM stock_stockreport WHERE domain=%s)", [self.name]
```

Comment:
can you put a comment explaining this isn't using the ORM for performance reasons?